frontend/src/components/Teacher/TeacherList.vue

```diff
@@ -47,15 +88,35 @@ const handleDelete = async teacherId => {
       <table class="rsprd-table">
         <thead>
           <tr class="rsprd-table__heading rsprd-table__row">
-            <th v-for="(value, key) in teacherFields" :key="key" class="rsprd-table__cell">{{ value }}</th>
+            <th v-for="field in teacherFields" :key="field.property" class="rsprd-table__cell">
+              {{ field.displayName }}
+            </th>
             <th class="rsprd-table__cell">Action</th>
           </tr>
         </thead>
         <tbody class="rsprd-table__body">
           <tr v-for="teacher in filteredTeachers" :key="teacher.id" class="rsprd-table__row">
-            <td v-for="(value, key) in teacherFields" :key="key" class="rsprd-table__cell">{{ teacher[key] }}</td>
+            <td v-for="field in teacherFields" :key="field.property" class="rsprd-table__cell">
+              <span v-if="isToEdit(teacher.id) && field.isEditable">
+                <input v-model="teacher[field.property]" type="text" class="rsprd-input--edit" />
+              </span>
+              <span v-else>
+                {{ teacher[field.property] }}
+              </span>
+            </td>
             <td class="rsprd-table__cell">
-              <button @click="handleDelete(teacher.id)" class="rsprd-btn rsprd-btn--delete">
+              <span v-if="isToEdit(teacher.id)">
```

Comment:
Nitpick: `div` is a more appropriate element to wrap groups of other elements. `span` should be more for text.